src/main/java/muramasa/antimatter/tile/TileEntityMachine.java

```diff
@@ -101,18 +97,22 @@ public TileEntityMachine(TileEntityType<?> tileType) {
     public TileEntityMachine(Machine<?> type) {
         this(type.getTileType());
         this.type = type;
+        coverHandler.setup(MachineCoverHandler::new);
+        itemHandler.setup(MachineItemHandler::new);
+        fluidHandler.setup(MachineFluidHandler::new);
+        energyHandler.setup(MachineEnergyHandler::new);
+        recipeHandler.setup(MachineRecipeHandler::new);
+        interactHandler.setup(MachineInteractHandler::new);
     }
 
     @Override
     public void onFirstTick() {
-        if (!coverHandler.isPresent() && has(COVERABLE)) coverHandler = Optional.of(new MachineCoverHandler(this));
-        if (!interactHandler.isPresent() && has(CONFIGURABLE)) interactHandler = Optional.of(new MachineInteractHandler(this));
-        //TODO: what are implications of this? just makes life easier
-        if (!itemHandler.isPresent() /*&& isServerSide()*/ && has(ITEM) && getMachineType().getGui().hasAnyItem(getMachineTier())) itemHandler = Optional.of(new MachineItemHandler(this));
-        if (!fluidHandler.isPresent() && isServerSide() && has(FLUID) && getMachineType().getGui().hasAnyFluid(getMachineTier())) fluidHandler = Optional.of(new MachineFluidHandler(this));
-        //Allow energyHandler on client? this should fix capabilities.
-        if (!energyHandler.isPresent() /*&& isServerSide()*/ && has(ENERGY)) energyHandler = Optional.of(new MachineEnergyHandler(this, has(GENERATOR)));
-        if (!recipeHandler.isPresent() && isServerSide() && has(RECIPE)) recipeHandler = Optional.of(new MachineRecipeHandler<>(this));
+        coverHandler.init();
```

Comment:
Everything should be done in the ctor...

Reply:
Any interactions with tile before onFirstTick will deadlock world

Reply:
You shouldn't be doing anything with the tile itself, all you're doing is initializing capabilities, and the ctor is the proper place to do it, and that isn't `onLoad`.

Reply:
ctor execute even earlier then Load(). tile, world not fully initialized. if I will init caps in ctor(), it will be the same as in onLoad()

Reply:
After the PR is pulled I'll show you what I mean.

Reply:
Oh, so it will be nice if you show the better approach on that. Thanks